odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -362,8 +363,12 @@ def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
 
         self._log_merge_operation(src_partners, dst_partner)
 
-        # delete source partner, since they are merged
-        src_partners.unlink()
+        # check parameter to see if it should delete or archive
+        delete_src_partners = str2bool(self.env["ir.config_parameter"].sudo().get_param("base.delete_merged_src_partners", True))
```

Comment:
Since deleting is the default, I think the parameter should be for archiving them.

Reply:
Done